### PR TITLE
FIX: fix format string float point issue

### PR DESCRIPTION
**Pull request by kbearXD**

### CODE REVIEW

**Review by c9s — DISMISSED** [review]

**Comment by c9s** [comment]
left for bohan to review this XD

File: pkg/fixedpoint/convert.go [hunk 1]
```diff
@@ -115,9 +115,50 @@ func (v Value) FormatString(prec int) string {
...
-	pow := math.Pow10(prec)
-	return strconv.FormatFloat(
-		math.Trunc(float64(v)/DefaultPow*pow)/pow, 'f', prec, 64)
+
+	u := int64(v)
+
+	// trunc precision
+	precDiff := DefaultPrecision - prec
+	if precDiff > 0 {
+		powDiff := int64(math.Pow10(precDiff))
```
Comment (gx578007) [inline pkg/fixedpoint/convert.go:124]: Better to use `math.Round` to convert float64 to int64.

File: pkg/fixedpoint/convert.go [hunk 2]
```diff
@@ -115,9 +115,50 @@ func (v Value) FormatString(prec int) string {
...
 	}
-	pow := math.Pow10(prec)
-	return strconv.FormatFloat(
-		math.Trunc(float64(v)/DefaultPow*pow)/pow, 'f', prec, 64)
+
+	u := int64(v)
+
+	// trunc precision
+	precDiff := DefaultPrecision - prec
+	if precDiff > 0 {
```
Comment (c9s) [inline pkg/fixedpoint/convert.go:123]: what about negative precDiff?

Comment (kbearXD) [reply pkg/fixedpoint/convert.go:123]: precDiff > 0 means the prec is larger than 8 and our precision is only 8 for fixedpoint.Value
As a result, we don't need to trunc it.

Comment (kbearXD) [reply pkg/fixedpoint/convert.go:123]: if prec = 5, precDiff = 8 - 5 = 3.
For 123456789 (1.23456789) => 123456000 (1.23456)

File: pkg/fixedpoint/convert_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		assert.Equal("-0.57000", s)
+	})
+
+	t.Run("Value(-123456789) with prec = 9, expected 1.23456789", func(t *testing.T) {
+		v := MustNewFromString("-1.23456789")
+		s := v.FormatString(9)
+		assert.Equal("-1.234567890", s)
+	})
+
+	t.Run("Value(1234567890) with prec = -1, expected 10", func(t *testing.T) {
```
Comment (gx578007) [inline pkg/fixedpoint/convert_test.go:48]: Could add another test with prec=-3 which expects "0"?

Comment (kbearXD) [reply pkg/fixedpoint/convert_test.go:48]: OK!

**Comment by c9s** [comment]
check failed

**Pull request merged**